### ISSUE TITLE
Tags aren't saved in XML when editing DayOne entries, and can't be searched for afterward

### ISSUE THREAD

**Issue by micahellison** [jrnl-org/jrnl#1049]
### Environment
  - Jrnl `--diagnostic` output: 
```
jrnl: v2.4.5
Python: 3.7.5 (tags/v3.7.5:5c02a39a0b, Oct 15 2019, 00:11:34) [MSC v.1916 64 bit (AMD64)]
OS: Windows 10
```
  - Install method:  develop branch

### Current Behavior

When you edit an entry in a DayOne journal and add a tag, the tag isn't saved in the DayOne XML tags section, and you can't find that entry by searching for that tag, either.

### Repro Steps

With a DayOne journal:
* Run `jrnl` and create an entry with no tags:
```
I am adding an entry
```
* Run `jrnl -1 --edit` and add a tag:
```
[2020-09-26 11:37 ] I am adding an entry.

Now here is a @newtag

# 64291462002711EBB2735882A894A7CA
```
* Search for the tag you added:
```
jrnl @newtag
```

The console outputs a couple line breaks and that's it. It should be showing that entry.

### Other Information

The tags are included as expected when creating new entries. It's just a problem with the edit functionality.

**Comment by jonakeys** [comment]
When trying to reproduce this with v3.0, the edited/added tag shows up as expected. Maybe it's fixed between v2.4.5 and now with other PR's?

**Comment by micahellison** [comment]
> When trying to reproduce this with v3.0, the edited/added tag shows up as expected. Maybe it's fixed between v2.4.5 and now with other PR's?

That's great news! Thanks for checking that.

I think this issue still needs a running test to be closed, just so we can be sure that we won't have a regression.

**Comment by jonakeys** [comment]
When reading more diligent I saw that it's specifically about DayOne entries and there the error still occurs indeed. Creating a PR for that.
